onnx/backend/test/case/node/negativeloglikelihoodloss.py

```diff
@@ -27,7 +27,8 @@ def compute_negative_log_likelihood_loss(input, target, weight=None, reduction='
         # setting mode='clip' to deal with ignore_index > C or < 0 cases.
         # when the target value is > C or < 0, it doesn't matter which value we are
         # taking in gather_weight, since it will be set to 0 in the following if-block
-        gather_weight = np.take(weight, target, mode='clip')
+        # use np.int_ to make it compatible with x86 machines
+        gather_weight = np.take(weight, np.array(target, dtype=np.int_), mode='clip')
```

Comment:
Sorry, I didn't understand this. What precision is this? Why don't we use explicit 32 or 64 bits like elsewhere in this PR?

Reply:
32-bit  machines somehow cannot handle numpy.int64 here so I used int_ (which means int32 for 32 machines and int64 for 64 machines). As you mentioned, using a fixed type here is better. I have changed it from int_ to int32. Thank you for the suggestion!